internal/k8s/advl4_controller.go

```diff
@@ -204,6 +205,73 @@ func (c *AviController) SetupNamespaceDeletionEventHandler(numWorkers uint32) {
 	c.informers.NSInformer.Informer().AddEventHandler(nsHandler)
 }
 
+func (c *AviController) AddNetworkInfoEventHandlers() {
+	fetchNST1LR := func(obj interface{}) (string, string, bool, error) {
+		var ns, t1lr string
+		ingCIDRFound := false
+		resourceObj := obj.(*unstructured.Unstructured)
+		ns = resourceObj.GetNamespace()
+		topology, ok := resourceObj.Object["topology"]
+		if !ok {
+			err := fmt.Errorf("topology key not found in namespace network info object.")
+			return "", "", false, err
+		}
+		t1lr, ok = topology.(map[string]interface{})["gatewayPath"].(string)
+		if !ok || t1lr == "" {
+			err := fmt.Errorf("invalid gatewayPath found in namespace network info object.")
+			return "", "", false, err
+		}
+		ingCIDRs, ok := topology.(map[string]interface{})["ingressCIDRs"].([]interface{})
+		if ok && len(ingCIDRs) > 0 {
+			ingCIDRFound = true
+		}
+		return ns, t1lr, ingCIDRFound, nil
+	}
+
+	queue := utils.SharedWorkQueue().GetQueueByName(utils.ObjectIngestionLayer)
+	numWorkers := queue.NumWorkers
+
+	namespaceNetworkInfoHandler := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			utils.AviLog.Debugf("Namespace NetworkInfo Add")
+			ns, t1lr, ingCIDRPresent, err := fetchNST1LR(obj)
+			if err != nil {
+				utils.AviLog.Errorf(err.Error())
+			}
+			lib.SharedWCPLister().UpdateNamespaceTier1LrCache(ns, t1lr)
```

Comment:
Ideally, In ako, the create/update/delete of mappings are done in the Graph layer.

Reply:
Moved it to Graph Layer, but also had to move VipNetworkList setter/getter to utils as part of it.